cmd/juju/cloud/add.go

```diff
@@ -235,6 +264,45 @@ func (c *AddCloudCommand) runInteractive(ctxt *cmd.Context) error {
 	return nil
 }
 
+// addCertificate reads the cloud certificate file if available and adds the contents
+// to the byte slice with the appropriate key.
+func addCertificate(data []byte) (string, []byte, error) {
+	vals, err := ensureStringMaps(string(data))
+	if err != nil {
+		return "", nil, err
+	}
+	var certificate string
+	filename, ok := vals[cloud.CertFilenameKey].(string)
```

Comment:
This will panic if the `cloud.CertFilenameKey` is not there, because you're using the type assertion as well. If we're 100% sure that this will be fine, ignore my comment.

Reply:
Updating:  I can't guarantee the key in the map, but I can guarantee if the key in the map, the value is a string.